app/importers/rows/educator_row.rb

```diff
@@ -8,9 +8,13 @@ def self.build(row)
   def build
     return nil if row[:login_name].nil? || row[:login_name] == ''
 
-    educator = Educator.find_or_initialize_by(email: email)
+    email = email_from_row
+    login_name = login_name_from_row(email)
+    educator = Educator.find_or_initialize_by(login_name: login_name)
```

Comment:
This allows a bunch of edge cases for Somerville and New Bedford (where these aren't distinct concepts).  What happens if these values were different?  These also seem like places where tests across different district keys or `PerDistrict` return values would help make sure this works for the different cases (eg, first-time import and create, subsequence import and update).